front/app/modules/commercial/insights/admin/components/Tag/index.tsx

```diff
@@ -63,18 +63,26 @@ const TagContent = styled.div`
 
 const Tag = ({ label, onIconClick, status }: TagProps) => {
   const handleIconClick = (e: React.MouseEvent) => {
-    e.preventDefault();
     e.stopPropagation();
+    e.preventDefault();
     onIconClick();
   };
 
+  const handleEnterPress = (e: React.KeyboardEvent) => {
+    if (e.key === 'Enter') {
+      e.stopPropagation();
+      e.preventDefault();
+      onIconClick();
+    }
+  };
   return (
     <StyledTag status={status} data-testid="insightsTag">
       <TagContent>
         {label}
         <IconContainer
           onClick={handleIconClick}
-          role="button"
+          onKeyPress={handleEnterPress}
```

Comment:
Nitpicking Again, but the focus looks weird 
![image](https://user-images.githubusercontent.com/38424075/122539512-f94ff180-d027-11eb-8a84-10f661ea6756.png)

Reply:
Nice catch! Fixed :)